src/preprocess/esa_cci.py

```diff
@@ -35,43 +35,20 @@ def create_filename(self, netcdf_filepath: str,
             new_filename = f"{year}_{filename_stem}.nc"
         return new_filename
 
-    def remap_values_to_even_spaced_integers(self, ds: xr.Dataset,
-                                             remap: Optional[Dict] = None) -> xr.Dataset:
-        """https://stackoverflow.com/a/3404089/9940782"""
-        print("Remapping values to new codes")
-        assert 'lc_class' in ds.data_vars, "Should be run after preprocessing!"
-        new_ds = xr.ones_like(ds)
-
-        # extract and copy numpy array
-        array = ds.lc_class.values
-        new_array = copy(array)
-
-        if not remap:
-            # create remap dictionary
-            legend = pd.read_csv(self.raw_folder / self.dataset / 'legend.csv')
-            valid_vals = legend.code.values
-            remap_vals = np.arange(0, (len(valid_vals) * 10), 10)
-
-            assert isinstance(valid_vals, np.ndarray)
-            remap = dict(zip(valid_vals, remap_vals))
+    def _one_hot_encode(self, ds: xr.Dataset) -> xr.Dataset:
 
-            # save the new codes to pandas dataframe
-            legend['new_code'] = remap_vals
-            legend.to_csv(self.out_dir / 'legend.csv')
+        legend = pd.read_csv(self.raw_folder / self.dataset / 'legend.csv')
+        # no data should have a value of 0 in the legend
 
-        print("Performing the remap")
-        # perform the remap
-        for k, v in remap.items(): new_array[array == k] = v
-
-        # reassign values to new ds
-        new_ds['lc_class'] = (['time', 'lat', 'lon'], new_array)
-
-        return new_ds
+        for idx, row in legend.iterrows():
+            value, label = row.code, row.label
+            ds[f'{label}_one_hot'] = ds.lc_class.where(ds.lc_class == value, 0).clip(min=0, max=1)
```

Comment:
ooo nice use of `.clip`

Reply:
if `not one_hot_encode` do we use the old values? The remapping was mostly for plotting purposes

Reply:
yea - so not at the moment. I though it would actually be easier to go from the one hot encoding to a remapped value.

Should I preserve the remapping function?